src/relay/transforms/fold_constant.cc

```diff
@@ -120,6 +120,18 @@ class ConstantFolder : public MixedModeMutator {
     }
   }
 
+  Expr VisitExpr_(const IfNode* op) final {
+    auto new_cond = ExprMutator::VisitExpr(op->cond);
+    if (auto const_cond = new_cond.as<ConstantNode>()) {
+      if (reinterpret_cast<uint8_t*>(const_cond->data->data)[0]) {
```

Comment:
why not use a bool datatype for the cast?

Reply:
there is no representation of `bool*`, things need to be byte-addressable